tools/eslint-rules/tsconfig.json

```diff
@@ -1,7 +1,8 @@
 {
-  "extends": "../../tsconfig.json",
+  "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "module": "commonjs"
+    "module": "Node16",
+    "moduleResolution": "Node16"
```

Comment:
Why not Node18, Node20 or even NodeNext?

Reply:
I did it based on what was done in the Nx repo when `@typescript-eslint/utils` v6 was introduced: https://github.com/nrwl/nx/pull/19548/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519 
The next possible value is `NodeNext` and if you wish, we can try it out.

Reply:
After tests, value of the `module` and `moduleResolution` has been set to `NodeNext`